velox/expression/CastExpr.h

```diff
@@ -108,4 +108,38 @@ class CastExpr : public SpecialForm {
   const bool nullOnFailure_;
 };
 
+/// Custom operator for casts from and to custom types.
+class CastOperator {
+ public:
+  virtual ~CastOperator() = default;
+
+  /// Determine whether the cast operator supports casting the custom type to
+  /// the other type or vice versa.
+  virtual bool isSupportedType(const TypePtr& other) const = 0;
+
+  /// Cast an input vector to the custom type.
+  /// @param inputDecoded The decoded input vector
+  /// @param baseRows Non-null rows of the base vector of inputDecoded
+  /// @param fromType The type of the original input vector
+  /// @param result The output vector of the custom type (Caller must ensure it
+  /// is writable.)
+  virtual void castTo(
+      const DecodedVector& inputDecoded,
+      const SelectivityVector& baseRows,
+      const TypePtr& fromType,
```

Comment:
Is this needed? Isn't it the same as input->type()?

Reply:
Removed. Thanks!